src/expr/src/vector_op/agg/general_sorted_grouper.rs

```diff
@@ -53,41 +95,47 @@ impl EqGroups {
         use std::collections::BinaryHeap;
         let mut heap = BinaryHeap::new();
         for (ci, column) in columns.iter().enumerate() {
-            if let Some(ri) = column.0.get(0) {
+            if let Some(ri) = column.get_starting_indices().get(0) {
                 heap.push(Reverse((ri, ci, 0)));
             }
         }
         while let Some(Reverse((ri, ci, idx))) = heap.pop() {
-            if let Some(ri_next) = columns[ci].0.get(idx + 1) {
+            if let Some(ri_next) = columns[ci].get_starting_indices().get(idx + 1) {
                 heap.push(Reverse((ri_next, ci, idx + 1)));
             }
             if ret.last() == Some(ri) {
                 continue;
             }
             ret.push(*ri);
         }
-        EqGroups(ret)
+        // EqGroups(ret)
+        EqGroups::new(ret)
     }
 }
 
 /// `SortedGrouper` contains the state of a group column in the sort aggregate
 /// algorithm, just like `Aggregator` contains the state of an aggregate column.
 pub trait SortedGrouper: Send + 'static {
-    /// `split_groups` detects the `EqGroups` from the `input` array if appended
+    /// `detect_groups` detects the `EqGroups` from the `input` array if appended
     /// to current state. See the documentation of `EqGroups` to learn more.
     ///
     /// This is a `dry-run` and does not update its state yet, because it does not
     /// have grouping information from all group columns yet.
-    fn split_groups(&self, input: &ArrayImpl) -> Result<EqGroups>;
+    ///
+    /// `offset` offset to the input child chunk
+    fn detect_groups(&self, input: &ArrayImpl) -> Result<EqGroups>;
 
     /// `update_and_output_with_sorted_groups` updates with each subslice of the
     /// `input` array according to the `EqGroups`. Finished groups are outputted
     /// to `builder` immediately along the way. After this call, the internal state
     /// is about the last group which may continue in the next chunk. It can be
     /// obtained with `output` when there are no more upstream data.
+    ///
+    /// Siyuan: prepare sorted group to builder
```

Comment:
Not part of the rustdoc :)

```suggestion
    // TODO(Siyuan): prepare sorted group to builder
```